collaborative-groth16/src/groth16.rs

```diff
@@ -222,6 +248,7 @@ where
             roots[i] = roots[i - 1].square();
         }
         roots.reverse();
+        let (q, roots) = Self::root_of_unities(domain);
```

Comment:
if you call this here than all of the above should be deleted, no?

Reply:
deleted